File: client/src/types/blendFade2Config.ts
/** Blend fade 2 strategy config (1:1 with trading-cursor-models strategies/blend_fade2). */
export interface BlendFade2Config {
  lookback: number
  lookbackFast: number
  zThreshold: number
  minRangePct: number
  zReversal: boolean
  zFastMin: number
  rankConfirm: number
  zMax: number
  sessionUtcStart: number | null
  sessionUtcEnd: number | null
}

export const BLEND_FADE2_PRESET_ACTIVE: BlendFade2Config = {
  lookback: 50,
  lookbackFast: 20,
  zThreshold: 1.08,
  minRangePct: 0.0026,
  zReversal: false,
  zFastMin: 0.64,
  rankConfirm: 0,
  zMax: 0,
  sessionUtcStart: null,
  sessionUtcEnd: null,
}

/** Best PnL from search_tune (batch 2). */
export const BLEND_FADE2_PRESET_PNL_MAX: BlendFade2Config = {
  lookback: 48,
  lookbackFast: 18,
  zThreshold: 1.08,
  minRangePct: 0.0026,
  zReversal: false,
  zFastMin: 0.60,
  rankConfirm: 0,
  zMax: 0,
  sessionUtcStart: null,
  sessionUtcEnd: null,
}

export const DEFAULT_BLEND_FADE2_CONFIG: BlendFade2Config = BLEND_FADE2_PRESET_PNL_MAX

File: src/PolyTrader.Core/Strategy/BlendFade2Config.cs
namespace PolyTrader.Core.Strategy;

/// <summary>
/// Blend fade 2 — z-score mean-reversion on 5m closes
/// (1:1 with trading-cursor-models strategies/blend_fade2).
/// </summary>
public sealed class BlendFade2Config
{
    public int Lookback { get; init; } = 50;
    public int LookbackFast { get; init; } = 20;
    public double ZThreshold { get; init; } = 1.08;
    public double MinRangePct { get; init; } = 0.0026;
    public bool ZReversal { get; init; }
    public double ZFastMin { get; init; } = 0.64;
    public double RankConfirm { get; init; }
    public double ZMax { get; init; }
    public int? SessionUtcStart { get; init; }
    public int? SessionUtcEnd { get; init; }

    /// <summary>Same base as blend_fade blend_active; extras off.</summary>
    public static BlendFade2Config PresetActive() => new();

    /// <summary>Best PnL from search_tune (batch 2).</summary>
    public static BlendFade2Config PresetPnlMax() => new()
    {
        Lookback = 48,
        LookbackFast = 18,
        ZThreshold = 1.08,
        MinRangePct = 0.0026,
        ZFastMin = 0.60,
    };

    public static BlendFade2Config Default { get; } = PresetPnlMax();
}
